### PR TITLE
Use `etcd` module from Quorum with geth 1.9.7

### PR DESCRIPTION
**Pull request by ricardolyn**
### Summary

Quorum used a modified version of etcd 3.3.13. By moving to go modules, we then had to create our own fork with the modified version on https://github.com/ConsenSys/etcd/tree/v3.3.13-quorum197. 

In the future, the plan is to move to the latest version of etcd (without modifications) and adapt our raft code to use the latest API.

### CODE REVIEW

**Review by amalrajmani — APPROVED** [review]
LGTM

File: raft/constants.go [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+	etcdRaft "github.com/coreos/etcd/raft"
+)
+
 const (
 	//protocolName           = "raft"
 	//protocolVersion uint64 = 0x01
 
 	//raftMsg = 0x00
 
+	minterRole = etcdRaft.LEADER
```
Comment (nmvalera) [inline raft/constants.go:13]: I am not fan of re-declaring an etcd constant as a constant here (why not just using the constant itself?)

File: raft/handler.go [hunk 2]
```diff
@@ -668,24 +669,34 @@ func (pm *ProtocolManager) isVerifier(rid uint16) bool {
 	return false
 }
 
-func (pm *ProtocolManager) handleRoleChange(raftRole etcdRaft.StateType) {
+func (pm *ProtocolManager) handleRoleChange(roleC <-chan interface{}) {
```
Comment (nmvalera) [inline raft/handler.go:672]: Why not using and `chan etcd.StateType`?

**Review by nmvalera — APPROVED** [review]
LGTM

File: raft/constants.go (hunk 1 above)
Comment (ricardolyn) [reply raft/constants.go:13]: @amalrajmani only applied the code on quorum-geth-1.9.7. we need in the future to re-do this using the right StateType API on this issue: https://github.com/ConsenSysQuorum/quorum/issues/189

File: raft/handler.go (hunk 2 above)
Comment (ricardolyn) [reply raft/handler.go:672]: @amalrajmani only applied the code on quorum-geth-1.9.7. we need in the future to re-do this using the right StateType API on this issue: https://github.com/ConsenSysQuorum/quorum/issues/189

**Pull request merged**